tests/e2e/scale/test_osd_node_balancing.py

```diff
@@ -0,0 +1,193 @@
+"""
+Test osd node balancing by adding nodes and osds and checking their distribution
+"""
+import logging
+from ocs_ci.framework.testlib import scale, E2ETest, ignore_leftovers
+from ocs_ci.ocs import constants, machine, ocp
+from ocs_ci.ocs.node import add_new_node_and_label_it, get_osd_running_nodes, get_nodes
+from ocs_ci.ocs.resources import storage_cluster
+from ocs_ci.utility.utils import (
+    ceph_health_check,
+    TimeoutSampler,
+    ocsci_log_path,
+)
+
+REPORT = "Node_and_Osd_distribution_report.txt"
```

Comment:
I believe the intention here is to retain the stages of the test in a file, but it would be great if this is done in the form of logging and as part of the flow of the test. 
@vasukulkarni need your thoughts here pls.

Reply:
IMO, it will be good idea to push the report as doc to the elastic-search server.

Reply:
Pushing the report to the elastic search server is now a new git issue that I will work on (4185).

Reply:
We should definitely add the logs or it becomes difficult to debug,  i think i also need to go over with warren to understand it better, i am missing few parts of the logic.

Reply:
This new version uses the elastic search database to save these results